### PR TITLE
Derek furst/multipe components

### PR DESCRIPTION
**Pull request by DerekFurstPitt**
Implemented multiple components endpoint (companion endpoint to the entity-api components endpoint). Makes a call to entity api's endpoint where all validation is done. If it returns a 200, it uses ingest file helper to create directories for each of the datasets created.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -696,6 +696,49 @@ def create_datastage():
...
+    entity_type = 'dataset'
+    try:
+        component_request = request.json
+        auth_helper = AuthHelper.configured_instance(app.config['APP_CLIENT_ID'], app.config['APP_CLIENT_SECRET'])
+        auth_tokens = auth_helper.getAuthorizationTokens(request.headers)
+        if isinstance(auth_tokens, Response):
+            return(auth_tokens)
+        elif isinstance(auth_tokens, str):
+            token = auth_tokens
+        elif 'nexus_token' in auth_tokens:
```
Comment (yuanzhou) [inline src/app.py:712]: The code is probably copied from the old `POST /datasets` function. As a matter of fact, `nexus_token` had been deprecated by Globus. Just remove this `elif` block here. We don't touch the old code though.

File: src/app.py [hunk 2]
```diff
@@ -696,6 +696,49 @@ def create_datastage():
...
+            token = auth_tokens['nexus_token']
+        else:
+            return(Response("Valid nexus auth token required", 401))
```
Comment (yuanzhou) [inline src/app.py:715]: Let's use `globus groups token` in the message

**Review by yuanzhou — CHANGES_REQUESTED** [review]
@DerekFurstPitt to support this new endpoint, you'll also need to define it in the "old gateway": https://github.com/hubmapconsortium/gateway. Please branch off `main` and PR back to `dev-integrate` with adding the new endpoint to `api_endpoints.<dev|test|stage|prod>.json` file under the ingest-api section. Require the read group `5777527e-ec11-11e8-ab41-0af86edb4424` to pass auth should be fine, the entity-api checks the group write permission with AWS API Gateway more thoroughly in the subsequent call.

**Comment by DerekFurstPitt** [comment]
@yuanzhou The requested changes have been made, changes from main have been pulled in. After long discussions with @maxsibilla we arrived at how we should handle the symlink stuff. It works in my local testing, but I had to make heavy changes to the request as well as my configs because I don't have a top level /hive/ directory and can't create files off the root like that. I want to test it with the entity-api piece that's on dev to verify that it still works properly.

**Review by yuanzhou — CHANGES_REQUESTED** [review]
@DerekFurstPitt we only need to do all the `dataset_link_abs_dir` validations upfront before making a call to entity-api. As I think about this endpoint and the one in entity-api a bit more, we don't even need to send over the `dataset_link_abs_dir` field to entity-api and do extra validation there then return it back again. This field is only relevant to ingest-api, entity-api shouldn't even care about it.

@maxsibilla @shirey what do you think?

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**